cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/MeasureStratifierTest.java

```diff
@@ -0,0 +1,290 @@
+                .firstGroup()
+                .stratifierById("stratifier-1")
+                .stratumCount(2)
+                .stratum(isUnfinished)
+                .firstPopulation()
+                .hasCount(9)
+                .up()
+                .up()
+                .stratum(notUnfinished)
+                .firstPopulation()
+                .hasCount(1)
+                .up()
+                .up()
+                .up()
+                .stratifierById("stratifier-2")
+                .stratumCount(2)
+                .stratum(mCC)
+                .firstPopulation()
+                .hasCount(5)
+                .up()
+                .up()
+                .stratum(fCC)
+                .firstPopulation()
+                .hasCount(5)
+                .up()
+                .up()
+                .up()
+                .up()
+                .report();
+    }
+    /**
+     * Boolean Basis Measure with Stratifier defined without an 'id' populated. Result should throw an error.
+     */
+    @Test
+    void cohortBooleanNoIdStrat() {
+        try {
+            given.when().measureId("CohortBooleanStratNoId").evaluate().then().report();
+            fail();
+        } catch (NullPointerException e) {
+            assertTrue(e.getMessage().contains("id is required on all Elements of type: Measure.group.stratifier"));
+        }
+    }
+    /**
+     * Boolean Basis Measure with Stratifier defined that produces resource based results.
+     * 9 Encounters are returned as a result of this stratifier expression. One stratum per unique 'Encounter' found.
+     * Example: Encounter/patient-1-encounter-1 as a stratum value returned.
+     * This is validating that even though the Measure is Boolean basis, a stratifier expression that produces results of a different basis IS possible if the expression can be evaluated in the correct context.
+     * If this expression produced both 'boolean' AND 'resource' results then it should not be allowed.
+     */
+    @Test
+    void cohortBooleanDifferentTypeStrat() {
+        given.when()
+                .measureId("CohortBooleanStratDifferentType")
+                .evaluate()
+                .then()
+                .firstGroup()
+                .firstStratifier()
+                .stratumCount(9) // one for each encounter
+                .up()
+                .up()
+                .report();
+    }
+    /**
+     * Boolean Basis Measure with Stratifier defined as a component.
+     * MultiComponent stratifiers blend results of multiple criteria (Gender of Patient and Payer, instead of just one or the other)
+     * This is allowed within the specification, but is not currently implemented
+     */
+    @Test
+    void cohortBooleanComponentStrat() {
+        try {
+            given.when()
+                    .measureId("CohortBooleanStratComponent")
+                    .evaluate()
+                    .then()
+                    .report();
+            fail();
```

Comment:
Might be a good idea to add a brief message to fail() explaning the expectation.